pyo3-derive-backend/src/method.rs

```diff
@@ -139,6 +140,15 @@ impl<'a> FnSpec<'a> {
             };
         }
 
+        if let FnType::ClassAttribute = &fn_type {
+            if self_.is_some() || !arguments.is_empty() {
+                return Err(syn::Error::new_spanned(
+                    name,
+                    "Class attribute methods cannot take arguments",
```

Comment:
It may be helpful to optionally allow one argument, of type `Python`, for those who want to create python types as class attributes. See `pymethod/impl_wrap_getter` as an example of how we allow this for getters.

One caveat I am not sure about though: as we're currently in the middle of creating a type object, is it safe to run arbitrary Python code?

Reply:
> is it safe to run arbitrary Python code?

Maybe this code can cause SIGSEGV.
```rust
#[pymethods]
impl MyClass {
    #[classattr]
    fn foo() -> MyClass { ... } 
}
```

I'll open a small PR to prevent this.

Reply:
@kngwyu is your conclusion that this is safe?